cpp/src/io/parquet/delta_binary.cuh

```diff
@@ -219,6 +219,8 @@ struct delta_binary_decoder {
 
     // need to account for the first value from header on first pass
     if (current_value_idx == 0) {
+      // make sure all threads access current_value_idx above before incrementing
+      __syncwarp();
```

Comment:
This sync seems unnecessary.  If t0 is the only thread that ever modifies `current_value_idx` there's never a case where it can be out of sync with itself.  And having all the other threads wait on the existing __syncwarp() takes care of this particular modification.

Reply:
but can thread 0 modify `current_value_idx` before thread 31 does the test for `current_value_idx == 0`? If that happens, and `value_count = 1`, then thread 0 could return and thread 31 continue, no?

Reply:
All threads read `current_value_idx` in line 218 and 221. For example, if `current_value_idx == 0` and `value_count == 1`, thread 0 can bump `current_value_idx` to 1 before other threads read `current_value_idx` in line 218 or 221. This would cause some threads to take the incorrect path.

Reply:
I don't think that can happen.  Threads in a warp don't run independently of each other.  Everyone will evaluate `if(current_value_idx == 0)` at the same time (except exited threads). But once we have the branch `if (lane_id == 0) { current_value_idx++; }` that write to shared could be incomplete by the time everyone does the final if.

Reply:
> Everyone will evaluate if(current_value_idx == 0) at the same time

I don't think that's guaranteed on recent archs.

Reply:
After some internal discussions I think this is fine.  It certainly doesn't hurt anything.